service/rpc/server.go

```diff
@@ -60,16 +67,21 @@ func (s *ServerImpl) Run() error {
 		return err
 	}
 
+	rpcs := grpc.NewServer()
+	rpcs.Register(s.s)
+
 	go func() {
-		c, err := s.s.listener.Accept()
-		if err != nil {
-			panic(err)
-		}
 		defer s.s.listener.Close()
-
-		rpcs := grpc.NewServer()
-		rpcs.Register(s.s)
-		rpcs.ServeCodec(jsonrpc.NewServerCodec(c))
+		for {
+			c, err := s.s.listener.Accept()
+			if err != nil {
+				panic(err)
```

Comment:
Also, if we go with that model then we should also place the `close(s.s.stopChan)` in a defer in case `Detach` returns an error, we would still want to close the listener and cleanup connections.